field_encrypt.module

```diff
@@ -131,44 +124,115 @@ function field_encrypt_form_field_add_form_builder($entity_type, \Drupal\field\E
+  field_encrypt_entity_encrypt($entity);
+}
+
+/**
+ * Encrypt the saved entity.
  *
  * @param \Drupal\Core\Entity\EntityInterface $entity
  */
-function field_encrypt_entity_presave(\Drupal\Core\Entity\EntityInterface $entity) {
-  if (!($entity instanceof Drupal\Core\Entity\ContentEntityInterface)) {return;}
-  /**
-   * @var $field_encrypt_process_entities \Drupal\field_encrypt\FieldEncryptProcessEntities
-   */
+function field_encrypt_entity_encrypt(Drupal\Core\Entity\EntityInterface $entity) {
+  if (($entity instanceof Drupal\field_encrypt\Entity\EncryptedFieldValueInterface) || !($entity instanceof Drupal\Core\Entity\ContentEntityInterface)) {
+    return;
+  }
+  // Check if this entity has fields that should be encrypted.
   $field_encrypt_process_entities = \Drupal::service('field_encrypt.process_entities');
-  $field_encrypt_process_entities->encryptEntity($entity);
+  if (!$field_encrypt_process_entities->checkEntity($entity)) {
+    return;
+  }
+
+  // Set a flag to only perform the encryption once.
+  if ($entity->encrypting != TRUE) {
+    $entity->encrypting = TRUE;
+    // Our encryption save action should never trigger a new revision.
+    if ($entity->getEntityType()->hasKey('revision')) {
+      $entity->setNewRevision(FALSE);
+    }
+    $entity->save();
+  }
+}
+
+/**
+ * Implements hook_entity_presave().
+ *
+ * Encrypt entity fields before they are saved.
+ */
+function field_encrypt_entity_presave(\Drupal\Core\Entity\EntityInterface $entity) {
+  if (($entity instanceof Drupal\field_encrypt\Entity\EncryptedFieldValueInterface) || !($entity instanceof Drupal\Core\Entity\ContentEntityInterface)) {
+    return;
+  }
+
+  // Only encrypt when the correct flag is set.
+  if ($entity->encrypting == TRUE) {
+    /* @var $field_encrypt_process_entities \Drupal\field_encrypt\FieldEncryptProcessEntities */
+    $field_encrypt_process_entities = \Drupal::service('field_encrypt.process_entities');
+    $field_encrypt_process_entities->encryptEntity($entity);
+  }
 }
 
 /**
- * Decrypt fields before they are rendered.
+ * Implement hook_entity_storage_load().
  *
- * @param $entities
- * @param $entity_type
+ * Decrypt entity fields when loading entities.
  */
-function field_encrypt_entity_load($entities, $entity_type) {
+function field_encrypt_entity_storage_load($entities, $entity_type) {
   /**
    * @var $field_encrypt_process_entities \Drupal\field_encrypt\FieldEncryptProcessEntities
    */
   $field_encrypt_process_entities = \Drupal::service('field_encrypt.process_entities');
 
-  foreach($entities as &$entity) {
-    if (!($entity instanceof Drupal\Core\Entity\ContentEntityInterface)) {continue;}
-    $field_encrypt_process_entities->decryptEntity($entity);
+  foreach ($entities as &$entity) {
+    if (!($entity instanceof Drupal\Core\Entity\ContentEntityInterface)) {
+      continue;
+    }
+    if (!$field_encrypt_process_entities->checkEntity($entity)) {
+      continue;
```

Comment:
Stylistically, I'm not a huge fan of the "continue" syntax.

Ideally, we just wrap this in one checking function